tests/integration/test_core.py

```diff
@@ -117,6 +116,121 @@ async def test_mysql_relation(ops_test: pytest_operator.plugin.OpsTest, applicat
     )
 
 
+@pytest.mark.asyncio
+async def test_openstack_object_storage_plugin(
+    ops_test: pytest_operator.plugin.OpsTest,
+    application_name,
+    default_admin_password,
+    unit_ip_list,
+    openstack_environment,
+):
+    """
+    arrange: after charm deployed, db relation established and openstack swift server ready.
+    act: update charm configuration for openstack object storage plugin.
+    assert: openstack object storage plugin should be installed after the config update and
+        WordPress openstack swift object storage integration should be set up properly.
+        After openstack swift plugin activated, an image file uploaded to one unit through
+        WordPress media uploader should be accessible from all units.
+    """
+    assert ops_test.model
+    # create the swift client
+    swift_conn = swiftclient.Connection(
+        authurl=openstack_environment["OS_AUTH_URL"],
+        auth_version="3",
+        user=openstack_environment["OS_USERNAME"],
+        key=openstack_environment["OS_PASSWORD"],
+        os_options={
+            "user_domain_name": openstack_environment["OS_USER_DOMAIN_ID"],
+            "project_domain_name": openstack_environment["OS_PROJECT_DOMAIN_ID"],
+            "project_name": openstack_environment["OS_PROJECT_NAME"],
+        },
+    )
+    container_exists = True
+    container = "WordPress"
+    # check if the swift container exists, if the container exists,
+    # remove every object and delete the container
+    try:
+        swift_conn.head_container(container)
+    except swiftclient.exceptions.ClientException as exc:
+        if exc.http_status == 404:
+            container_exists = False
+        else:
+            raise exc
+    if container_exists:
+        for swift_object in swift_conn.get_container(container, full_listing=True)[1]:
+            swift_conn.delete_object(container, swift_object["name"])
+        swift_conn.delete_container(container)
```

Comment:
Can't you use the `delete` method from the SwiftService that could do all this in 1 operation ?

Reply:
Also just saw that the SwiftService instance is create right after this

Reply:
That's really convenient. Fixed, thanks!